File: javascript/191.number-of-1-bits.js
/*
 * @lc app=leetcode id=191 lang=javascript
 *
 * [191] Number of 1 Bits
 */

// @lc code=start
/**
 * @param {number} n - a positive integer
 * @return {number}
 */
var hammingWeight = function (n) {
  let w = 0;
  while (n) {
    if (n & 1) w++;
    n >>>= 1;
  }
  return w;
};
// @lc code=end

File: rust/191.number-of-1-bits.rs
/*
 * @lc app=leetcode id=191 lang=rust
 *
 * [191] Number of 1 Bits
 */

// @lc code=start
impl Solution {
    pub fn hammingWeight(n: u32) -> i32 {
        // Solution 1:
        return n.count_ones() as i32;

        // Solution 2:
        // let (mut result, mut x) = (0, n);
        // for i in 0..32 {
        //     result += (x & 1) as i32;
        //     x >>= 1;
        // }
        // return result;
    }
}
// @lc code=end
